src/subdomains/core/liquidity-management/adapters/balances/blockchain.adapter.ts

```diff
@@ -154,6 +157,34 @@ export class BlockchainAdapter implements LiquidityBalanceIntegration {
     const blockchain = assets[0].blockchain;
     const client = this.evmRegistryService.getClient(blockchain);
 
+    const coinBalance = await client.getNativeCoinBalanceByAlchemy();
+    const tokenBalances = await client.getTokenBalancesByAlchemy();
+
+    const tokenToBalanceMap = new Map<string, number>(
+      tokenBalances.filter((t) => t.symbol).map((t) => [t.symbol.toUpperCase(), t.balance ? Number(t.balance) : 0]),
+    );
+
+    for (const asset of assets) {
+      try {
+        const balance =
+          asset.type === AssetType.COIN
+            ? client.fromWeiAmount(coinBalance)
+            : tokenToBalanceMap.get(asset.dexName.toUpperCase()) ?? 0;
+
+        this.balanceCache.set(asset.id, balance);
+      } catch (e) {
+        this.logger.error(`Failed to update liquidity management balance for ${asset.uniqueName}:`, e);
+        this.invalidateCacheFor([asset]);
+      }
```

Comment:
The try-catch doesn't make much sense here. Should be around lines 160-161 (and invalidate cache for all assets).